solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -181,17 +191,28 @@ contract WalletRegistry is Ownable {
 
     /// @notice Requests a new wallet creation.
     /// @dev Can be called only by the owner of wallets.
-    ///      It starts a new DKG process.
+    ///      It locks the DKG and request a new random relay entry. It expects
+    ///      that the DKG process will be started once a new random relay entry
+    ///      gets generated.
     function requestNewWallet() external {
         require(msg.sender == walletOwner, "Caller is not the Wallet Owner");
 
         dkg.lockState();
-        // TODO: When integrating with the Random Beacon move `dkg.start` to a
-        // callback function. We need each DKG to be started with a unique
-        // and fresh relay entry.
-        dkg.start(
-            uint256(keccak256(abi.encodePacked(relayEntry, block.number)))
+
+        randomBeacon.requestRelayEntry(this);
+    }
+
+    /// @notice A callback that is executed once a new random relay entry gets
+    ///         generated. It starts the DKG process.
+    /// @dev Can be called only by the random beacon contract.
+    /// @param randomRelayEntry Random relay entry.
+    function __beaconCallback(uint256 randomRelayEntry, uint256) external {
```

Comment:
What is the gas cost of this function? Are we sure it's going to fit under the random beacon's callback gas limit? Could be worth to mention it in `@dev` doc for this function.

Reply:
In tests, it's `77307` on avarage.

Reply:
Currently, RandomBeacon's `callbackGasLimit` is set to `50000`.

Reply:
:untada:

Reply:
e01090f05ccca3ddf50dfda58ca92a17f817a688

Reply:
Just double-checking, is 56k enough when the callback is called for the very first time? There is a gas cost difference when we modify a value that is already a non-default one and when we change a value from a default one to something else.

Reply:
We'll be always setting a value over a default (`0`). For the first time, the struct is initialized with a default, and before the next run we clean up the seed with `delete` https://github.com/keep-network/keep-core/blob/15022421ca49ac41811cd6b9d26cb9b160aa0d80/solidity/ecdsa/contracts/libraries/EcdsaDkg.sol#L512.